tests/react/optimization.test.tsx

```diff
@@ -271,3 +271,78 @@ it('no extra rerenders after commit with derived atoms (#1213)', async () => {
   })
   expect(renderCount1).toBe(renderCount1AfterCommit)
 })
+
+const readAtomStateTestCase = (enableOptimization: boolean) => {
+  const queryParams = atom({
+    key: '123',
+    query: 'foobar',
+  })
+  const authAtom = atom({
+    isLoggedInUser: true,
+  })
+  const hasProfileAtom = atom((get) => {
+    const auth = get(authAtom)
+    return auth.isLoggedInUser
+  })
+  const mockDep1 = atom(1)
+  const mockDep2 = atom(2)
+
+  const queryAtom = atom((get) => {
+    get(queryParams)
+    get(authAtom)
+    get(hasProfileAtom)
+    get(mockDep1)
+    get(mockDep2)
+    return {
+      data: {
+        alert: 'This is an alert',
+        content: ['This is the page content'],
+        cart: {
+          items: [],
+        },
+      },
+    }
+  })
+  const alertAtom = atom((get) => {
+    return get(queryAtom).data.alert
+  })
+  const contentAtom = atom((get) => {
+    return get(queryAtom).data.content
+  })
+  const cartAtom = atom((get) => {
+    return get(queryAtom).data.cart
+  })
+
+  const store = createStore({ unmountedCacheEnabled: enableOptimization })
+  const MyPage = () => {
+    useAtom(alertAtom)
+    useAtom(contentAtom)
+    useAtom(cartAtom)
+    return 'Hello world'
+  }
+  render(
+    <Provider store={store}>
+      <MyPage />
+    </Provider>,
+  )
+
+  store.dev_print_call_counts?.()
+  store.dev_clear_call_counts?.()
+}
+it('Avoid redundant readAtomState while unmount', () => {
```

Comment:
in this test case we didn't expect anything. just run the code? I think we should add some counter to make sure behavior is stable